client/src/extension.ts

```diff
@@ -59,6 +60,87 @@ export function activate(context: ExtensionContext) {
 
 	// Start the client. This will also launch the server
 	client.start();
+	
+	const ctrl = tests.createTestController('perlNavigator', 'Perl Navigator');
+	context.subscriptions.push(ctrl);
+
+	const fileTag = new TestTag('file');
+	// Run profile for .t files marked with above tag
+	ctrl.createRunProfile('Suite', TestRunProfileKind.Run, (request) => {
+		let fileItems: ReadonlyArray<TestItem>;
+		// Play button clicked on a specific test
+		if (request.include && request.include.length > 0) {
+			fileItems = request.include;
+		}
+		// Run tests button for all
+		else {
+			fileItems = [...ctrl.items].map(([_, testItem]) => testItem);
+		}
+
+		for (const fileItem of fileItems) {
+			fileItem.canResolveChildren = true;
+			const run = ctrl.createTestRun(request, fileItem.uri.fsPath);
+			run.started(fileItem);
+			// Extract via Test2::API to JSON
+			const execPromise = new Promise((resolve, reject) => {
+				child_process.exec(`perl -e '
```

Comment:
We'll need to update this to use `settings.perlPath` instead of "perl" somehow.

This is the way I call external perl scripts in the server side:
```
const process = async_execFile(settings.perlPath, settings.perlParams.concat(tidyParams), { timeout: 25000, maxBuffer: 20 * 1024 * 1024 }); 
```